crates/red_knot_python_semantic/src/types.rs

```diff
@@ -107,11 +107,29 @@ fn widen_type_for_undeclared_public_symbol<'db>(
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) enum SymbolLookup {
+    Internal,
+    External,
+}
```

Comment:
I wonder if the `internal` and `external` should be an internal type only to simplify the implementation but that we otherwise use `internal_global_symbol`, `external_global_symbol` methods instead (internal global symbol seems weird :)).

Reply:
I thought of doing that but quickly realize the `internal_global_symbol` name issue. For now, I just went with it to complete it and finish the implementation. I'll do some name / structural changes now with the benchmarking.